docs/semantics.md

```diff
@@ -402,8 +412,8 @@ Transient errors can occur in distributed systems like Cloud Storage, such as ne
 ## Missing features
 
 Not all of the usual file system features are supported. Most prominently:
-- Renaming directories is by default not supported. A directory rename cannot be performed atomically in Cloud Storage and would therefore be arbitrarily expensive in terms of Cloud Storage operations, and for large directories would have high probability of failure, leaving the two directories in an inconsistent state.
-- However, if your application can tolerate the risks, you may enable renaming directories in a non-atomic way, by setting ```--rename-dir-limit```. If a directory contains fewer files than this limit and no subdirectory, it can be renamed.
+- Renaming directories is by default not supported in non-hierarchical buckets. A directory rename cannot be performed atomically in these buckets and would therefore be arbitrarily expensive in terms of Cloud Storage operations, and for large directories would have high probability of failure, leaving the two directories in an inconsistent state. This can be resolved using cloud storage buckets with hierarchical namespace enabled. HNS enabled buckets support atomic folder renames and are fully compatible with Cloud Storage FUSE, offering seamless integration.
+- However, if your application is using non-hierarchical buckets and can tolerate the risks, you may enable renaming directories in a non-atomic way, by setting ```--rename-dir-limit```. If a directory contains fewer files than this limit and no subdirectory, it can be renamed.
```

Comment:
This comment is not related to this PR but it was be good to add to complete this guidance. 
"by setting rename-dir-limit to number of maximum files you expect to be under a directory" and/or link to the public documentation where the rename dir limit is documented.

Reply:
Change non-hierarchical to flat

Reply:
Done